audits/viewport-meta-tag/index.js

```diff
@@ -28,22 +28,41 @@ class ViewportMetaTagTest {
       return Promise.reject('No data provided.');
     }
 
-    if (typeof inputs.dom !== 'function') {
-      return Promise.reject('No DOM Parser provided.');
+    if (typeof inputs.driver === 'undefined') {
+      return Promise.reject('No Driver provided.');
     }
 
-    let domParser = inputs.dom;
-    let viewport = domParser('meta[name="viewport"]');
+    let driver = inputs.driver;
+    let ret = {};
 
-    // If there's a viewport return a score of 1.
-    // TODO(paullewis): make this test more nuanced.
-    if (typeof viewport !== 'undefined' && viewport !== null) {
-      return Promise.resolve(true);
-    }
+    return driver
+      .requestTab(inputs.url)
+      // make this not totally stupid.
+      .then((instance) => {
+        return driver.evaluateScript(findMetaViewport)
+            .then((obj) => {
+              if (obj.type === "object" && obj.subtype === "null")
```

Comment:
should this just structurally be set up to describe what a valid viewport is (allowing easy extension in the future as that changes) and then return false otherwise? Or should it be that we enumerate what makes up a valid viewport, enumerate what makes an invalid viewport, and throw on everything else (as you're doing here) to make sure we know we aren't handling all possible permutations?